sebs/gcp/gcp.py

```diff
@@ -374,24 +364,82 @@ def update_function(self, function: Function, code_package: Benchmark):
             )
         self.logging.info("Published new function code and configuration.")
 
-    def update_function_configuration(self, function: Function, benchmark: Benchmark):
+    def _update_envs(self, full_function_name: str, envs: dict) -> dict:
+
+        get_req = (
+            self.function_client.projects().locations().functions().get(name=full_function_name)
+        )
+        response = get_req.execute()
+
+        # preserve old variables while adding new ones.
+        # but for conflict, we select the new one
+        if "environmentVariables" in response:
+            envs = {**response["environmentVariables"], **envs}
+
+        return envs
+
+    def _generate_function_envs(self, code_package: Benchmark) -> dict:
```

Comment:
What other env vars do we think this function will have to support in the future (so apart from NoSQL-related)?

Reply:
@oanarosca If we use Redis for something, we also need to pass Reddis access details.